src/main/scala/org/apache/daffodil/tdml/processor/IBM_DFDL.scala

```diff
@@ -510,9 +506,9 @@ sealed class IBMTDMLResult(diags: Seq[IBMTDMLDiagnostic]) {
 
   def isValidationError: Boolean = diagnostics.exists { _.getType() == DFDLDiagnosticType.VALIDATIONERROR }
 
-  def getDiagnostics: Seq[Diagnostic] = diagnostics
+  def getDiagnostics: java.util.List[api.Diagnostic] = diagnostics.asJava
 
-  def addDiagnostic(diag: Diagnostic): Unit = { diagnostics = diag.asInstanceOf[IBMTDMLDiagnostic] +: diagnostics }
+  def addDiagnostic(diag: iapi.Diagnostic): Unit = { diagnostics = diag.asInstanceOf[IBMTDMLDiagnostic] +: diagnostics }
```

Comment:
This is probably just a bug in Daffodil that this needs to be iapi.Diagnostic. Similarly, we probably shouldn't require the use of the iapi.DaffodilSchemaSource/URISchemaSource/etc--that's all internal and really shouldn't be needed to implement a tdml processor.

That said, that's fairly minor and there are so few tdml processors that's it's probably not worth blocking this. But can you create a Daffodil ticket to update the TDML API to not depend on internal classes and we can get it fixed in a future release?

Reply:
Created https://issues.apache.org/jira/browse/DAFFODIL-3033 to track the issue